test/ClangDisplayContext/test.cpp

```diff
@@ -0,0 +1,142 @@
+// This test is line- and column-sensitive, so test commands are at the bottom.
+// Tests are identical with Analysis/bug_hash_test.cpp
+
+constexpr int clang_analyzer_hashDump(int) { return 5; }
+
+void function(int) {
+  clang_analyzer_hashDump(5);
+}
+
+namespace {
+void variadicParam(int, ...) {
+  clang_analyzer_hashDump(5);
+}
+} // namespace
+
+constexpr int f() {
+  return clang_analyzer_hashDump(5);
+}
+
+namespace AA {
+class X {
+  X() {
+    clang_analyzer_hashDump(5);
+  }
+
+  static void static_method() {
+    clang_analyzer_hashDump(5);
+    variadicParam(5);
+  }
+
+  void method() && {
+    struct Y {
+      inline void method() const & {
+        clang_analyzer_hashDump(5);
+      }
+    };
+
+    Y y;
+    y.method();
+
+    clang_analyzer_hashDump(5);
+  }
+
+  void OutOfLine();
+
+  X &operator=(int) {
+    clang_analyzer_hashDump(5);
+    return *this;
+  }
+
+  operator int() {
+    clang_analyzer_hashDump(5);
+    return 0;
+  }
+
+  explicit operator float() {
+    clang_analyzer_hashDump(5);
+    return 0;
+  }
+};
+} // namespace AA
+
+void AA::X::OutOfLine() {
+  clang_analyzer_hashDump(5);
+}
+
+void testLambda() {
+  []() {
+    clang_analyzer_hashDump(5);
+  }();
+}
+
+template <typename T>
+void f(T) {
+  clang_analyzer_hashDump(5);
+}
+
+template <typename T>
+struct TX {
+  void f(T) {
+    clang_analyzer_hashDump(5);
+  }
+};
+
+template <>
+void f<long>(long) {
+  clang_analyzer_hashDump(5);
+}
+
+template <>
+struct TX<long> {
+  void f(long) {
+    clang_analyzer_hashDump(5);
+  }
+};
+
+template <typename T>
+struct TTX {
+  template<typename S>
+  void f(T, S) {
+    clang_analyzer_hashDump(5);
+  }
+};
+
+// RUN: mkdir -p %T
```

Comment:
There is `%t` which is a unique temporary file for a test. Could you use that instead of `mkdir` and `%T`?